RFS/src/main/java/org/opensearch/migrations/bulkload/common/OpenSearchClient.java

```diff
@@ -405,10 +407,12 @@ public Mono<BulkResponse> sendBulkRequest(String indexName, List<BulkDocSection>
                     var successfulDocs = resp.getSuccessfulDocs();
                     successfulDocs.forEach(docsMap::remove);
                     log.atWarn()
-                        .setMessage("After bulk request on index '{}', {} more documents have succeed, {} remain")
+                        .setMessage("After bulk request attempt {} on index '{}', {} more documents have succeeded, {} remain. The error response message was: {}")
+                        .addArgument(attempt)
                         .addArgument(indexName)
                         .addArgument(successfulDocs::size)
                         .addArgument(docsMap::size)
+                        .addArgument(response.body)
```

Comment:
By logging the full body we could be putting megabytes of response data into these logs multiple times, what do you think about truncating what is being output?

Reply:
Good idea, I've added some truncation logic here and some test cases around it